tools/dockerfile/grpc_objc/generate_boringssl_prefix_header/generate_boringssl_prefix_header.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+[ $# == 1 ] || { echo "Usage: generate_boringssl_prefix_header.sh <boringssl_commit>" ; exit 1 ; }
+
+git clone -n https://github.com/google/boringssl.git
```

Comment:
seems completely unnecessary. You have the right commit of boringssl in third_party/boringssl already so why this?

Reply:
Part of the reason was I did not find a way to cleanly copy both `/tools/dockerfile/grpc_objc/generate_boringssl_prefix_header/generate_boringssl_prefix_header.sh` and `/third_party/boringssl` into the container. To make it possible I probably need to set the Docker build context to be gRPC's root dir, but that means transferring the entire grpc directory to Docker when running `docker build`, which takes much more time than `git clone` when I tried it.

And also for the same sanity reason as above, it does not cost too much to get a clean version of BoringSSL.